### PR TITLE
Hint at draw_without_rendering() in Text.get_window_extent

### PR DESCRIPTION
**Pull request by timhoffm**

### CODE REVIEW

File: lib/matplotlib/text.py [hunk 1]
```diff
@@ -882,8 +882,9 @@ def get_window_extent(self, renderer=None, dpi=None):
             A renderer is needed to compute the bounding box.  If the artist
             has already been drawn, the renderer is cached; thus, it is only
             necessary to pass this argument when calling `get_window_extent`
-            before the first `draw`.  In practice, it is usually easier to
-            trigger a draw first (e.g. by saving the figure).
+            before the first draw.  In practice, it is usually easier to
```
Comment (timhoffm) [inline lib/matplotlib/text.py:885]: The `draw` link would go to [Text.draw](https://matplotlib.org/devdocs/api/text_api.html#matplotlib.text.Text.draw) which is more or less an internal API and not helpful here (at worst the user might try and call that). It's better to not link to anything here.

File: lib/matplotlib/text.py [hunk 2]
```diff
@@ -882,8 +882,9 @@ def get_window_extent(self, renderer=None, dpi=None):
...
+            trigger a draw first, e.g. by calling
+            `~.Figure.draw_without_rendering` or ``plt.show()``.
```
Comment (timhoffm) [inline lib/matplotlib/text.py:887]: Saving is usually not a good way to trigger a draw in this case. You'll likely want to modify the figure somehow based on the extent.

Comment (tacaswell) [reply lib/matplotlib/text.py:887]: I'm 50/50 about including `plt.show()` as

```python
plt.ion()
for j in range(5):
    fig, ax = plt.subplots()
    plt.show()
    sleep(5)
```

will not spin the GUI event loop (and hence in some cases not get the actual render to happen) until after the loop is done.

**Review by tacaswell — APPROVED** [review]
Left a minor comment, but :+1: on it going in as is.

File: lib/matplotlib/text.py (hunk 2 above)
Comment (jklymak) [reply lib/matplotlib/text.py:887]: OK, but 99.999% of cases, a draw will occur!

**Review by jklymak — APPROVED** [review]

**Pull request merged**